servers/rendering/renderer_rd/shaders/canvas_uniforms_inc.glsl

```diff
@@ -178,12 +176,43 @@ transforms;
 
 /* SET3: Texture */
 
-layout(set = 3, binding = 0) uniform texture2D color_texture;
-layout(set = 3, binding = 1) uniform texture2D normal_texture;
-layout(set = 3, binding = 2) uniform texture2D specular_texture;
-layout(set = 3, binding = 3) uniform sampler texture_sampler;
+#extension GL_EXT_nonuniform_qualifier : require
```

Comment:
I have incorporated @clayjohn's fixes for the artefacts observed on AMD devices

Reply:
For future reference, this is a safe change to require since GLSLang supports this extension. GLSLang will output some SPIRV hints which should be ignored by devices that don't support it